http-core/src/main/scala/org/apache/pekko/http/javadsl/settings/ConnectionPoolSettings.scala

```diff
@@ -37,7 +36,7 @@ abstract class ConnectionPoolSettings private[pekko] () { self: ConnectionPoolSe
   def getMaxRetries: Int = maxRetries
   def getMaxOpenRequests: Int = maxOpenRequests
   def getPipeliningLimit: Int = pipeliningLimit
-  def getMaxConnectionLifetime: JDuration = maxConnectionLifetime.asJava
+  def getMaxConnectionLifetime: JDuration = JDuration.ofNanos(maxConnectionLifetime.toNanos)
```

Comment:
can be toJava

Reply:
scala.jdk.DurationConverters only works with FiniteDuration - maxConnectionLifetime is not FiniteDuration

Reply:
Basically we have a bug in pekko.util.JavaDurationConverters.
It allows you to compile:
```
import pekko.util.JavaDurationConverters._
scala.concurrent.duration.Duration.Inf.asJava
```

But this will fail at runtime because pekko.util.JavaDurationConverters calls toNanos on the scala duration and this fails with
```
java.lang.IllegalArgumentException: toNanos not allowed on infinite Durations
	at scala.concurrent.duration.Duration$Infinite.fail(Duration.scala:219)
```

I'm going to park this PR for a while - and concentrate on fixing the bug.
I think pekko.util.JavaDurationConverters should return the max possible Java Duration because Java Duration does not support infinite.